cmd/helmExecute.go

```diff
@@ -68,9 +67,35 @@ func runHelmExecute(helmCommand string, helmExecutor kubernetes.HelmExecutor) er
 		if err := helmExecutor.RunHelmPackage(); err != nil {
 			return fmt.Errorf("failed to execute helm package: %v", err)
 		}
-	case "push":
-		if err := helmExecutor.RunHelmPush(); err != nil {
-			return fmt.Errorf("failed to execute helm push: %v", err)
+	case "publish":
+		if err := helmExecutor.RunHelmPublish(); err != nil {
+			return fmt.Errorf("failed to execute helm publish: %v", err)
+		}
+	default:
+		if err := runHelmExecuteDefault(config, helmExecutor); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func runHelmExecuteDefault(config helmExecuteOptions, helmExecutor kubernetes.HelmExecutor) error {
+	if config.LintFlag {
+		if err := helmExecutor.RunHelmLint(); err != nil {
+			return fmt.Errorf("failed to execute helm lint: %v", err)
+		}
+	}
+
+	if config.PackageFlag {
+		if err := helmExecutor.RunHelmPackage(); err != nil {
```

Comment:
before packaging, we may update the version based on the config parameter `artifactVersion` - if set

This should help to use the proper version of the chart as well since we cannot promote the same version multiple times ...

Reply:
done via https://github.com/SAP/jenkins-library/pull/3627